### PR TITLE
NCL-823 Build Agent routing

### PR DESCRIPTION
**Pull request by slaskawi**

### CODE REVIEW

File: termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    public RunningBuild startProjectBuild(BuildConfiguration buildConfiguration, final RunningEnvironment runningEnvironment)
+            throws BuildDriverException {
+
+        return new RunningBuild() {
+
+            @Override
+            public void monitor(Consumer<CompletedBuild> onComplete, Consumer<Exception> onError) {
+                onComplete.accept(new CompletedBuild() {
+                    @Override
+                    public BuildDriverStatus getCompleteStatus() {
```
Comment (slaskawi) [inline termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java]: @matejonnet I think we should remove it.. There is already a build status in BuildResult.

File: termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    @Override
+                    public BuildDriverResult getBuildResult() throws BuildDriverException {
+                        return new BuildDriverResult() {
+                            @Override
+                            public String getBuildLog() throws BuildDriverException {
+                                return "Logs... logs everywhere...";
+                            }
+
+                            @Override
+                            public BuildDriverStatus getBuildDriverStatus() throws BuildDriverException {
```
Comment (slaskawi) [inline termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java]: This should not throw exception probably...

File: termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                                return BuildDriverStatus.SUCCESS;
+                            }
+
+                            @Override
+                            public RunningEnvironment getRunningEnvironment() {
```
Comment (slaskawi) [inline termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java]: Why are we returning running environment all over the place? I think it should be only returned by RunningBuild...

File: termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java (hunk 1 above)
Comment (matejonnet) [inline termd-build-driver/src/main/java/org/jboss/pnc/rest/notifications/model/TermdBuildDriver.java]: Looks like it can be removed as it is not used.

**Pull request merged**